Part of Emoji Reaction final design

jrainville opened the pull request:
### What does the PR do

Part of #18935

Contains multiple commits:

- [fix(EmojiPopup): add selected emoji to recentEmoji model](https://github.com/status-im/status-desktop/commit/822f40b8d29e0dd7ef4e05287605b4f29f39dbe2): Fixes an issue where recent emojis were not put in the model at all, so they never showed
- [refactor(EmojiPopup): only pass the needed props instead of all settings](https://github.com/status-im/status-desktop/commit/b35093d421a4ecdf9101e4fb3d373da415971003): Small clean up. Passes only the needed settings instead of all of them
- [feat(emoji): show recent emojis in the message context menu](https://github.com/status-im/status-desktop/commit/ac79391ef0168b8f32745166204ffadbaae21671): First part of the new design. Shows the 5 most recent emojis in the Message context menu. If no recent emojis, doesn't show the row.
- [fix(emojis): fix recentEmojis and skinTone not being properly saved](https://github.com/status-im/status-desktop/commit/492aa9a46998050eba30704a481abc96c87d228e): Fixes an issue with saving the recent emojis and the skin tone
- [feat(emojis): add button to open emoji popup in the message context menu](https://github.com/status-im/status-desktop/commit/87eaf84500cc6a0cef12dd867b7d6e8c4766329f): second part of the new design. Adds a button in the row in the Message context menu. When clicked, it opens the emoji popup
- [chore(emojis): remove no longer needed default emoji reactions model](https://github.com/status-im/status-desktop/commit/177463b5779b66bda5df7f5773c307b2d510839e): cleanup after the features to remove the no longer needed model

### Affected areas

- MessagesReactionRow
- MessageContextMenu
- AppMain
- MessageView

### Architecture compliance

- [x] I am familiar with the [application architecture](/docs/architecture.md) and agreed good practices.
My PR is consistent with this document: [QML Architecture Guidelines](/guidelines/QML_ARCHITECTURE_GUIDE.md)

### Screencapture of the functionality

[emoji-reaction-new-design-part-1.webm](https://github.com/user-attachments/assets/47d280ed-118d-4e6e-9f04-16bd7f0be2aa)

### Impact on end user

Nicer experience when using the emoji reactions in the message context menu + fixes the emoji settings

### How to test

- Use emojis in a message
- Right click a message and see that the latest emojis are there

### Risk 

Low

jrainville commented on ui/imports/shared/controls/chat/MessageReactionsRow.qml:
```diff
@@ -1,32 +1,100 @@
+            visible: index < d.recentEmojisModel.count
+            emojiId: visible ? emojiReaction.emoji.unicode : ""
+            // TODO not implemented yet. We'll need to pass this info
+            // reactedByUser: model.didIReactWithThisEmoji
```
Will be implemented in a follow up PR

caybro commented on ui/imports/shared/controls/chat/MessageReactionsRow.qml:
```diff
@@ -1,32 +1,94 @@
+            id: emojiReaction
+
+            required property int index
+            property var emoji: visible ? d.recentEmojisModel.get(index) : null
```
Yeah, maybe the `IndexFilter` slows things down; keep it at `d.recentEmojisModel.get(index)` if it works better for you

noeliaSD commented on ui/imports/shared/controls/chat/EmojiReaction.qml:
```diff
@@ -2,32 +2,33 @@ import QtQuick
+    StatusEmoji {
+        id: statusEmoji
         anchors.centerIn: parent
+        width: 20
```
Do we want a fixed size or something dynamic for the times coming? ;)

jrainville replied:
I followed the design, but you're right, the design was for Desktop only. What do you suggest to make it more dynamic?

micieslak replied:
Fixed value is far from ideal but the question is what should be the base for that size. Maybe it could be derived from some font size (from Theme of), done using `FontMentrics`?

noeliaSD commented on ui/imports/shared/controls/chat/MessageReactionsRow.qml:
```diff
@@ -1,32 +1,98 @@
     id: root
 
-    property var defaultEmojiReactionsModel
+    required property StatusEmojiModel emojiModel
```
It would be nice to describe which are the properties expected for this specific model

noeliaSD reviewed (commented):
Looks good overall!

1. I’m just thinking about how we manage the **recent emojis**, right now they’re duplicated both in `SQUtils.Emoji.emojisModel` and in `appMainLocalSettings.recentEmojis`, and we’re manually syncing one with the other. That feels a bit error-prone.
  Maybe we could simplify by having a single source of truth, for example, let `emojisModel` own the recents completely, loading them from `appMainLocalSettings` **on startup and then keeping the two bound** so updates automatically persist. That way we avoid having two live copies and reduce the chance of desync issues. WDYT?

2. One more architectural thought: why is the `emojisModel` living under `SQUtils.Emoji`?
  
 What I'm thinking is:
    - **SQUtils.Emoji** → source of truth for **static emoji data** + **stateless utilities** (catalog, metadata, search/indexing).
    - **App store (e.g., EmojiStore)** → **owns stateful, app-specific features** like _recents_, _favorites_, _custom ordering_, write-through persistence, etc.

  Concretely, we could:
1. Keep `SQUtils.Emoji` as a **data provider** (expose _allEmojisModel_ / search API, no app state).
2. Introduce an `EmojiStore` that wraps the provider and exposes:
  - _recentModel_, _favoritesModel_ (read-only models)
  - methods: toggleFavorite(emoji), clearRecents()..
  - persistence (load on startup from settings, boounded on change)
3. UI binds only to **EmojiStore.XXModel** ; **SQUtils.Emoji** never leaks _app state_.

 **Benefits:** clearer layering, easier testing/mocking, and we avoid coupling utils to product decisions (like which are the recents we keep).

WDYT?

caybro commented on ui/app/mainui/AppMain.qml:
```diff
@@ -1180,8 +1179,11 @@ Item {
+            skinColor: appMainLocalSettings.skinColor
             emojiModel: SQUtils.Emoji.emojiModel
+            onSetSkinColor: color => appMainLocalSettings.skinColor = color
+            onSetRecentEmojis: recentEmojis => appMainLocalSettings.recentEmojis = recentEmojis
```
I think it'd be cleaner to actually have a small `Settings {}` inside the popup and encapsulate the `skinColor` and `recentEmojis` entirely inside the popup. I'd consider that way cleaner than it is now; some of those settigns are handled inside the popup (in the opened/closed handlers), and some outside here

jrainville commented on ui/imports/shared/controls/chat/MessageReactionsRow.qml:
```diff
@@ -1,32 +1,98 @@
+    QtObject {
+        id: d
+
+        readonly property var recentEmojisModel: SortFilterProxyModel {
```
Yup, fixed in the [other PR](https://github.com/status-im/status-desktop/pull/19160)

jrainville commented:
@micieslak thanks for the reivew, I added a new commit to address your concerns. Some of the points are addresses in the [second PR](https://github.com/status-im/status-desktop/pull/19160) where I simplified the model passing.

I did see your comments there and I'll start addressing them right away!
